oak/client/sgx_application_client.cc

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak/client/sgx_application_client.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "asylo/grpc/auth/enclave_channel_credentials.h"
+#include "asylo/grpc/auth/enclave_credentials_options.h"
+#include "asylo/grpc/auth/null_credentials_options.h"
+//#include "asylo/grpc/auth/sgx_remote_credentials_options.h"
+#include "asylo/identity/descriptions.h"
+#include "asylo/identity/enclave_assertion_authority_configs.h"
+#include "asylo/identity/init.h"
+#include "asylo/identity/sgx/sgx_identity_util.h"
+#include "asylo/identity/util/sha256_hash.pb.h"
+#include "asylo/util/logging.h"
+#include "oak/client/authorization_bearer_token_metadata.h"
+#include "oak/client/policy_metadata.h"
+#include "oak/common/nonce_generator.h"
+#include "oak/common/policy.h"
+
+constexpr size_t kPerChannelNonceSizeBytes = 32;
+// Debug MRSIGNER value derived from the SGX test key (will change after changing SVN and PRODID).
+const char* kDebugMrSigner = "83d719e77deaca1470f6baf62a4d774303c899db69020f9c70ee1dfc08c7ce9e";
+
+// TODO: Use same function from asylo, when it will become public.
+bool Sha256HashFromHexString(const std::string& hex, asylo::Sha256HashProto* h) {
```

Comment:
Please clarify at least what the return value means. Even better, return a `Status` object with an appropriate error message instead.